src/renderer/graph/FrequencyResponseChart.tsx

```diff
@@ -21,41 +42,131 @@ const FrequencyResponseChart = () => {
     useAquaContext();
   const prevFilters = useRef<IFilter[]>([]);
   const prevFilterLines = useRef<ChartDataPoint[][]>([]);
+  const prevFreqPoints = useRef<ChartDataPointWithId[][]>([]);
 
   const {
     chartData,
+    pointData,
     autoPreAmpValue,
-  }: { chartData: ChartData[]; autoPreAmpValue: number } = useMemo(() => {
+  }: {
+    chartData: ChartCurveData[];
+    pointData: ChartPointsData[];
+    autoPreAmpValue: number;
+  } = useMemo(() => {
+    const freqData = filters.map(({ id, frequency }) => {
+      return { frequency, id };
+    });
+
+    const isFreqDiff =
+      prevFilters.current.length === freqData.length &&
+      !prevFilters.current.every(
+        ({ frequency }, i) => frequency === freqData[i].frequency
+      );
+
+    const hasNewFilter = prevFilters.current.length < freqData.length;
+
+    // Identify the index of each filter by id from previous render
+    const prevIndices: { [id: string]: number } = {};
+    prevFilters.current.forEach((f, i) => {
+      prevIndices[f.id] = i;
+    });
+
     // Update filter lines that have changed
-    const updatedFilterLines = filters.map((f, index) =>
-      index < prevFilters.current.length &&
-      isFilterEqual(f, prevFilters.current[index])
-        ? prevFilterLines.current[index]
-        : getFilterPoints(f)
-    );
+    const { data: updatedFilterLines, points: updatedFreqPoints } = filters
+      .map((f) => {
+        const prevIndex = prevIndices[f.id];
+        // New filters have no previous data
+        if (!prevIndex) {
+          return getFilterGraphData(f, freqData);
+        }
+
+        // Filter changes can impact both the curve and the points so update both
+        const hasFilterChanges = !isFilterEqual(
+          f,
+          prevFilters.current[prevIndex]
+        );
+
+        if (hasFilterChanges) {
+          return getFilterGraphData(f, freqData);
+        }
+
+        return {
+          data: prevFilterLines.current[prevIndex],
```

Comment:
Where do we actually compute prevFilterLines?

Reply:
They're updated within the `useMemo` for computing the chart data.